x/concentrated-liquidity/internal/math/tick_test.go

```diff
@@ -1,68 +1,219 @@
 package math_test
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/internal/math"
+	"github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/types"
 )
 
-func (suite *ConcentratedMathTestSuite) TestTickToSqrtPrice() {
+func (suite *ConcentratedMathTestSuite) TestTickToPrice() {
 	testCases := map[string]struct {
-		tickIndex         sdk.Int
-		sqrtPriceExpected string
+		tickIndex          sdk.Int
+		exponentAtPriceOne sdk.Int
+		expectedPrice      sdk.Dec
+		expectedError      error
 	}{
-		"positive tick index 1": {
-			tickIndex:         sdk.NewInt(85176),
-			sqrtPriceExpected: "70.710004849206120646",
-			// https://www.wolframalpha.com/input?i2d=true&i=Power%5B1.0001%2CDivide%5B85176%2C2%5D%5D
-		},
-		"positive tick index 2": {
-			tickIndex:         sdk.NewInt(86129),
-			sqrtPriceExpected: "74.160724590950847045",
-			// https://www.wolframalpha.com/input?i2d=true&i=Power%5B1.0001%2CDivide%5B86129%2C2%5D%5D
-		},
-		"negative tick index 1": {
-			tickIndex:         sdk.NewInt(-85176),
-			sqrtPriceExpected: "0.014142270278902791",
-			// https://www.wolframalpha.com/input?i2d=true&i=Power%5B1.0001%2CDivide%5B-85176%2C2%5D%5D
-		},
-		"negative tick index 2": {
-			tickIndex:         sdk.NewInt(-86129),
-			sqrtPriceExpected: "0.013484226394978088",
-			// https://www.wolframalpha.com/input?i2d=true&i=Power%5B1.0001%2CDivide%5B-86129%2C2%5D%5D
+		"One dollar increments at the ten thousands place: 1": {
+			tickIndex:          sdk.NewInt(400000),
+			exponentAtPriceOne: sdk.NewInt(-4),
+			expectedPrice:      sdk.NewDec(50000),
+		},
+		"One dollar increments at the ten thousands place: 2": {
+			tickIndex:          sdk.NewInt(400001),
+			exponentAtPriceOne: sdk.NewInt(-4),
+			expectedPrice:      sdk.NewDec(50001),
+		},
```

Comment:
What do you think of combining 1 and 2 test cases and then suite.Require.Equal(expecteddiff?)

Reply:
Wouldn't that make it a little harder to understand from an outsider reviewing the test? Maybe I am not understanding your point fully though